From 8d0665f7a6fd49719b8a671d303cd68c6faf4716 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 3 Dec 2017 11:58:15 +0000
Subject: [PATCH 3/5] meta: Stop overwriting utime/mtime of our files

This is now having an effect on packages where we need to see the mtime
change for triggers to work properly, such as for usysconf. Given that
this doesn't affect deltas, and only reproducibility, and that our
deltas kinda suck anyway, it's no big loss.

We'll revisit this when we replace eopkg with sol.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/metadata.py | 28 ++++++++++++++--------------
 1 file changed, 14 insertions(+), 14 deletions(-)

diff --git a/ypkg2/metadata.py b/ypkg2/metadata.py
index b453eee..ec86f29 100644
--- a/ypkg2/metadata.py
+++ b/ypkg2/metadata.py
@@ -422,20 +422,20 @@ def create_eopkg(context, gene, package, outputDir):
         orgname = os.path.join(context.get_install_dir(), finfo.path)
         orgname = orgname.encode('utf-8').decode('utf-8').encode("latin1")
 
-        if os.path.islink(orgname) and not os.path.isdir(orgname):
-            t = history_timestamp
-            cmd = "touch -d \"@{}\" -h \"{}\"".format(t, orgname)
-            try:
-                subprocess.check_call(cmd, shell=True)
-            except Exception as e:
-                console_ui.emit_warning("utime", "Failed to modify utime")
-                print("Reproducible builds will be affected: {}".format(e))
-        else:
-            try:
-                os.utime(orgname, (history_timestamp, history_timestamp))
-            except Exception as e:
-                console_ui.emit_warning("utime", "Failed to modify utime")
-                print("Reproducible builds will be affected: {}".format(e))
+        # if os.path.islink(orgname) and not os.path.isdir(orgname):
+        #     t = history_timestamp
+        #     cmd = "touch -d \"@{}\" -h \"{}\"".format(t, orgname)
+        #     try:
+        #         subprocess.check_call(cmd, shell=True)
+        #     except Exception as e:
+        #         console_ui.emit_warning("utime", "Failed to modify utime")
+        #         print("Reproducible builds will be affected: {}".format(e))
+        # else:
+        #     try:
+        #         os.utime(orgname, (history_timestamp, history_timestamp))
+        #     except Exception as e:
+        #         console_ui.emit_warning("utime", "Failed to modify utime")
+        #         print("Reproducible builds will be affected: {}".format(e))
 
         pkg.add_to_install(orgname, finfo.path)
 
-- 
2.15.1

